Up2dateService/Up2dateService/SetupManager/SetupManager.cs

```diff
@@ -150,28 +224,38 @@ private void SafeAddOrUpdatePackage(Package package)
 
         private async Task InstallPackages(IEnumerable<Package> packagesToInstall)
         {
-            foreach (Package inPackage in packagesToInstall)
+            await Task.Run(() =>
             {
-                if (!Path.GetExtension(inPackage.Filepath).Equals(MsiExtension, StringComparison.InvariantCultureIgnoreCase)) continue;
-
-                List<Package> lockedPackages = SafeGetPackages();
+                foreach (string inPackage in packagesToInstall.Select(inPackage => inPackage.Filepath))
+                {
+                    if (!AllowedExtensions.Contains(Path.GetExtension(inPackage),
+                            StringComparer.InvariantCultureIgnoreCase)) continue;
 
-                Package package = lockedPackages.FirstOrDefault(p => p.Filepath.Equals(inPackage.Filepath, StringComparison.InvariantCultureIgnoreCase));
-                if (package.Status == PackageStatus.Unavailable) continue;
+                    var lockedPackages = SafeGetPackages();
 
-                package.ErrorCode = 0;
-                package.Status = PackageStatus.Installing;
+                    Package package = lockedPackages.FirstOrDefault(p =>
+                        p.Filepath.Equals(inPackage, StringComparison.InvariantCultureIgnoreCase));
+                    if (package.Status == PackageStatus.Unavailable) continue;
 
-                SafeUpdatePackage(package);
+                    if (package.ProductName.Contains(Up2DateChocoId)) continue;
+                    package.ErrorCode = 0;
+                    package.Status = PackageStatus.Installing;
 
-                int result = await InstallPackageAsync(package, CancellationToken.None);
+                    SafeUpdatePackage(package);
+                    int result = string.Equals(Path.GetExtension(package.Filepath),
+                        NugetExtension,
+                        StringComparison.InvariantCultureIgnoreCase)
+                        ? (int)InstallChocoNupkg(package)
```

Comment:
InstallChocoNupakg shall only be called if choco is installed. If not - should report appropriate error.

Reply:
Added filtration Inside InstallChocoNupkg if choco is not installed the function immediatly exits with InstallPackageStatus.ChocoNotInstalled